force-app/main/domain/Households.cls

```diff
@@ -101,18 +103,48 @@ public inherited sharing class Households extends fflib_SObjects2 {
 
     /*******************************************************************************************************
     * @description Constructor
-    * @param accountIds Account Ids to initialize the Contact Repository with
+    * @param accountIds Account Ids to use for selector query
     */
-    public Households(Set<Id> accountIds) {
+    @TestVisible
+    private Households(Set<Id> accountIds) {
         super((List<Object>) householdSelector.findByIds(accountIds));
     }
 
     /*******************************************************************************************************
     * @description Constructor
-    * @param accountIds Account Ids to initialize the Contact Repository with
+    * @param accounts List of SObject records to use for domain constructor
     */
-    public Households(List<Account> accounts) {
-        super(accounts);
+    @TestVisible
+    private Households(List<SObject> records) {
+        super(records, Account.SObjectType);
+    }
+
+    public static IHouseholds newInstance(List<SObject> records) {
+        Application.Domain.replaceWith(Account.SObjectType, IHouseholds.class, Households.Constructor.class);
+        return (IHouseholds) Application.Domain.newInstance(records, Account.SObjectType);
+    }
+
+    public static IHouseholds newInstance(Set<Id> recordIds) {
+        Application.Domain.replaceWith(Account.SObjectType, IHouseholds.class, Households.Constructor.class);
+        return (IHouseholds) Application.Domain.newInstance(householdSelector.findByIds(recordIds), Account.SObjectType);
+    }
+
+    public IHouseholds getHouseholds() {
+        return (IHouseholds) getRecords(
```

Comment:
I believe we should be returning the an instance of `Households` here because we can not cast the List<SObject> directly to `IHouseholds`. It should look like 
`public IHouseholds getHouseholds() {
        return new Households((List<SObject>) getRecords(
            isHouseholdAccount()
        ));
    }`